src/platform/common/platform/fileSystem.node.ts

```diff
@@ -88,34 +66,70 @@ export class FileSystem extends FileSystemBase implements IFileSystemNode {
         return Array.isArray(found) ? found : [];
     }
 
-    areLocalPathsSame(path1: string, path2: string): boolean {
-        return arePathsSame(path1, path2);
+    async writeLocalFile(filename: string, text: string | Buffer): Promise<void> {
+        await fs.ensureDir(path.dirname(filename));
+        return fs.writeFile(filename, text);
     }
 
-    public async createLocalDirectory(path: string): Promise<void> {
-        await fs.ensureDir(path);
+    override async readFile(uri: Uri): Promise<string> {
+        if (isLocalFile(uri)) {
```

Comment:
Optimized the code to always use fs modules instead of VS Code API when dealing with local fs resources